experimental/generation/generator/packages/library/templates/generator.lg

```diff
@@ -342,6 +334,79 @@ ${setOrPushOperation()},
     "value": "=@${entity}"          
 }```
 
+# isWithMinMax()
+- IF: ${propertySchema.minimum || propertySchema.maximum}
+    -```
+{
+    "$kind": "Microsoft.IfCondition",
+    "condition": "@${entity}.number < dialogClass.schema.properties['${property}'].minimum",
+    "actions": [
+        {
+            "$kind": "Microsoft.SendActivity",
+            "activity": "\${${property}_IsTooSmall(@${entity}.number)}"
+        }
+    ],
+    "elseActions": [
+        {
+            "$kind": "Microsoft.IfCondition",
+            "condition": "@${entity}.number > dialogClass.schema.properties['${property}'].maximum",
+            "actions": [
+                {
+                    "$kind": "Microsoft.SendActivity",
+                    "activity": "\${${property}_IsTooBig(@${entity}.number)}"
+                }
+            ],
+            "elseActions": [
+                {
+                    "$kind": "Microsoft.SendActivity",
+                    "activity": "\${confirmationText('${property}',  @${entity}.number)}"
+                },
+            ${setOrPush()}
+            ]
+        }
+    ]
+}``` 
+- ELSEIF: ${propertySchema.items.minimum || propertySchema.items.maximum}
+    - ```
+{
+    "$kind": "Microsoft.IfCondition",
+    "condition": "@${entity}.number < dialogClass.schema.properties['${property}'].items.minimum",
```

Comment:
>dialogClass.schema.properties['${property}'].items.minimum [](start = 38, length = 58)

Pretty sure you don't need this--items properties are promoted up to parent property.  I also think we rely on the parent min/max in the *IsTooX templates.

Reply:
Should the condition be @${entity}.number < dialogClass.schema.properties['${property}'].items.minimum ？

Reply:
You don't need to anything with respect to items.minimum/maximum

---
In reply to: [565886958](https://github.com/microsoft/BotBuilder-Samples/pull/3042#discussion_r565886958) [](ancestors = 565886958)